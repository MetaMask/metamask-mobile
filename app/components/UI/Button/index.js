import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, StyleSheet } from 'react-native';
import GenericButton from '../GenericButton'; // eslint-disable-line import/no-unresolved
import { useTheme } from '../../../util/theme';

const createStyles = (colors) =>
  StyleSheet.create({
    button: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: colors.primary.default,
      paddingVertical: 10,
      paddingHorizontal: 15,
      height: 40,
      borderRadius: 4,
    },
  });

/**
 * UI component that wraps GenericButton
 * which renders the appropiate UI elements for each platform (android & iOS)
 */
const Button = (props) => {
  const { colors } = useTheme();
  const styles = createStyles(colors);

  return (
    <GenericButton onPress={props.onPress} style={[styles.button, props.style]}>
      {props.children}
    </GenericButton>
  );
};

Button.propTypes = {
  /**
   * Children components of the Button
   * it can be a text node, an image, or an icon
   * or an Array with a combination of them
   */
  children: PropTypes.any,
  /**
   * Styles to be applied to the Button
   */
  style: ViewPropTypes.style,
  /**
   * Function to be called on press
   */
  onPress: PropTypes.func,
};

export default Button;
