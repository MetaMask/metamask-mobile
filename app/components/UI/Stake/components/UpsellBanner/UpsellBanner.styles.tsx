import { StyleSheet } from 'react-native';
import { Theme } from '../../../../../util/theme/models';

const upsellBannerStylesheet = (params: { theme: Theme }) => {
  const { theme } = params;
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      backgroundColor: colors.background.alternative,
      borderRadius: 8,
      gap: 8,
      alignItems: 'center',
      paddingVertical: 24,
      paddingHorizontal: 16,
    },
  });
};

export default upsellBannerStylesheet;
