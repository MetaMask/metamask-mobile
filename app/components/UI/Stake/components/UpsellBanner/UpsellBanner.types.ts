export enum UPSELL_BANNER_VARIANTS {
  HEADER = 'HEADER',
  BODY = 'BODY',
}

interface UpsellBannerBaseProps {
  primaryText: string;
  secondaryText: string;
  tertiaryText: string;
  endAccessory?: React.ReactNode;
}

export type UpsellBannerHeaderProps = UpsellBannerBaseProps;

export type UpsellBannerBodyProps = UpsellBannerBaseProps & {
  onTooltipPress: () => void;
};

export type UpsellBannerProps =
  | (UpsellBannerHeaderProps & { variant: UPSELL_BANNER_VARIANTS.HEADER })
  | (UpsellBannerBodyProps & {
      variant: UPSELL_BANNER_VARIANTS.BODY;
    });
