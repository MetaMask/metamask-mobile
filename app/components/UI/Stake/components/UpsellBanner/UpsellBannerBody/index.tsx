import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import styleSheet from './UpsellBannerBody.styles';
import { UpsellBannerBodyProps } from '../UpsellBanner.types';
import { useStyles } from '../../../../../hooks/useStyles';
import Text, {
  TextVariant,
  TextColor,
} from '../../../../../../component-library/components/Texts/Text';
import Icon, {
  IconName,
  IconColor,
  IconSize,
} from '../../../../../../component-library/components/Icons/Icon';

const UpsellBannerBody = ({
  primaryText,
  secondaryText,
  tertiaryText,
  onTooltipPress,
  endAccessory,
}: UpsellBannerBodyProps) => {
  const { styles } = useStyles(styleSheet, {});

  return (
    <View style={styles.container}>
      <View style={styles.left}>
        <Text variant={TextVariant.HeadingMD}>{primaryText}</Text>
        <Text variant={TextVariant.DisplayMD} color={TextColor.Success}>
          {secondaryText}
        </Text>
        <View style={styles.tooltipContainer}>
          <Text variant={TextVariant.BodyMD} color={TextColor.Alternative}>
            {tertiaryText}
          </Text>
          <TouchableOpacity onPress={onTooltipPress}>
            <Icon
              name={IconName.Info}
              size={IconSize.Sm}
              color={IconColor.Alternative}
            />
          </TouchableOpacity>
        </View>
      </View>
      <View style={styles.right}>
        {React.isValidElement(endAccessory) && endAccessory}
      </View>
    </View>
  );
};

export default UpsellBannerBody;
