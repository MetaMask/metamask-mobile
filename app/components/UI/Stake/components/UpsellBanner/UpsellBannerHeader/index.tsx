import React from 'react';
import { View } from 'react-native';
import { UpsellBannerHeaderProps } from '../UpsellBanner.types';
import styleSheet from './UpsellBannerHeader.styles';
import { useStyles } from '../../../../../hooks/useStyles';
import Text, {
  TextVariant,
  TextColor,
} from '../../../../../../component-library/components/Texts/Text';

const UpsellBannerHeader = ({
  primaryText,
  secondaryText,
  tertiaryText,
  endAccessory,
}: UpsellBannerHeaderProps) => {
  const { styles } = useStyles(styleSheet, {});

  return (
    <View style={styles.container}>
      <Text variant={TextVariant.HeadingMD}>{primaryText}</Text>
      <Text variant={TextVariant.DisplayMD} color={TextColor.Success}>
        {secondaryText}
      </Text>
      <Text color={TextColor.Alternative}>{tertiaryText}</Text>
      {React.isValidElement(endAccessory) && endAccessory}
    </View>
  );
};

export default UpsellBannerHeader;
