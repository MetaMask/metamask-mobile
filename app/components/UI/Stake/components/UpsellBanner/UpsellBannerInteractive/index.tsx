import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import styleSheet from './UpsellBannerInteractive.styles';
import { UpsellBannerInteractiveProps } from '../UpsellBanner.types';
import { useStyles } from '../../../../../hooks/useStyles';
import Button, {
  ButtonVariants,
} from '../../../../../../component-library/components/Buttons/Button';
import Text, {
  TextVariant,
  TextColor,
} from '../../../../../../component-library/components/Texts/Text';
import Icon, {
  IconName,
  IconColor,
  IconSize,
} from '../../../../../../component-library/components/Icons/Icon';

const UpsellBannerInteractive = ({
  primaryText,
  secondaryText,
  tertiaryText,
  buttonLabel,
  onButtonPress,
  onTooltipPress,
}: UpsellBannerInteractiveProps) => {
  const { styles } = useStyles(styleSheet, {});

  return (
    <View style={styles.container}>
      <View style={styles.left}>
        <Text variant={TextVariant.HeadingMD}>{primaryText}</Text>
        <Text variant={TextVariant.DisplayMD} color={TextColor.Success}>
          {secondaryText}
        </Text>
        <View style={styles.tooltipContainer}>
          <Text variant={TextVariant.BodyMD} color={TextColor.Alternative}>
            {tertiaryText}
          </Text>
          <TouchableOpacity onPress={onTooltipPress}>
            <Icon
              name={IconName.Info}
              size={IconSize.Sm}
              color={IconColor.Alternative}
            />
          </TouchableOpacity>
        </View>
      </View>
      <View style={styles.right}>
        <Button
          variant={ButtonVariants.Secondary}
          label={buttonLabel}
          onPress={onButtonPress}
        />
      </View>
    </View>
  );
};

export default UpsellBannerInteractive;
