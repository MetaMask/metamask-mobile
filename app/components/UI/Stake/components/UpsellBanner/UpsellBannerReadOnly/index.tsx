import React from 'react';
import { View } from 'react-native';
import { UpsellBannerReadOnlyProps } from '../UpsellBanner.types';
import styleSheet from './UpsellBannerReadOnly.styles';
import { useStyles } from '../../../../../hooks/useStyles';
import Text, {
  TextVariant,
  TextColor,
} from '../../../../../../component-library/components/Texts/Text';

const UpsellBannerReadOnly = ({
  primaryText,
  secondaryText,
  tertiaryText,
}: UpsellBannerReadOnlyProps) => {
  const { styles } = useStyles(styleSheet, {});

  return (
    <View style={styles.container}>
      <Text variant={TextVariant.HeadingMD}>{primaryText}</Text>
      <Text variant={TextVariant.DisplayMD} color={TextColor.Success}>
        {secondaryText}
      </Text>
      <Text color={TextColor.Alternative}>{tertiaryText}</Text>
    </View>
  );
};

export default UpsellBannerReadOnly;
