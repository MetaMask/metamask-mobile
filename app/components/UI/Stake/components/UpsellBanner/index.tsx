import React from 'react';
import {
  UPSELL_BANNER_VARIANTS,
  UpsellBannerBodyProps,
  UpsellBannerProps,
  UpsellBannerHeaderProps,
} from './UpsellBanner.types';
import UpsellBannerBody from './UpsellBannerBody';
import UpsellBannerHeader from './UpsellBannerHeader';

const UpsellBanner = ({
  variant = UPSELL_BANNER_VARIANTS.HEADER,
  ...props
}: UpsellBannerProps) => {
  switch (variant) {
    case UPSELL_BANNER_VARIANTS.BODY:
      return <UpsellBannerBody {...(props as UpsellBannerBodyProps)} />;
    case UPSELL_BANNER_VARIANTS.HEADER:
    default:
      return <UpsellBannerHeader {...(props as UpsellBannerHeaderProps)} />;
  }
};

export default UpsellBanner;
