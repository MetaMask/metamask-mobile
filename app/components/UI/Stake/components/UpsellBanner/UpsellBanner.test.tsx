import React from 'react';
import UpsellBanner from '.';
import {
  UPSELL_BANNER_VARIANTS,
  UpsellBannerProps,
} from './UpsellBanner.types';
import renderWithProvider from '../../../../../util/test/renderWithProvider';
import Button, {
  ButtonVariants,
} from '../../../../../component-library/components/Buttons/Button';

describe('UpsellBanner', () => {
  const baseProps = {
    primaryText: 'you could earn',
    secondaryText: '$454',
    tertiaryText: 'per year on your tokens',
    endAccessory: (
      <Button
        label={'Earn 4.5%'}
        variant={ButtonVariants.Secondary}
        onPress={jest.fn()}
      />
    ),
  };

  describe('UpsellBannerHeader variant', () => {
    it('render matches screenshot', () => {
      const props: UpsellBannerProps = {
        variant: UPSELL_BANNER_VARIANTS.HEADER,
        ...baseProps,
      };

      const { toJSON } = renderWithProvider(<UpsellBanner {...props} />);

      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('UpsellBannerBody variant', () => {
    it('render matches screenshot', () => {
      const props: UpsellBannerProps = {
        variant: UPSELL_BANNER_VARIANTS.BODY,
        onTooltipPress: jest.fn(),
        ...baseProps,
      };

      const { toJSON } = renderWithProvider(<UpsellBanner {...props} />);

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
