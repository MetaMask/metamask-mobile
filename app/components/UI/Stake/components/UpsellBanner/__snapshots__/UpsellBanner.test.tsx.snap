// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpsellBanner UpsellBannerBody variant render matches screenshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f2f4f6",
      "borderRadius": 8,
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#141618",
          "fontFamily": "EuclidCircularB-Bold",
          "fontSize": 18,
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 24,
        }
      }
    >
      you could earn
    </Text>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#1c8234",
          "fontFamily": "EuclidCircularB-Bold",
          "fontSize": 32,
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 40,
        }
      }
    >
      $454
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#6a737d",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
      >
        per year on your tokens
      </Text>
      <TouchableOpacity
        onPress={[MockFunction]}
      >
        <SvgMock
          color="#6a737d"
          height={16}
          name="Info"
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
          width={16}
        />
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      {
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[MockFunction]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderColor": "#0376c9",
          "borderRadius": 20,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#0376c9",
            "fontFamily": "EuclidCircularB-Medium",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
      >
        Earn 4.5%
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`UpsellBanner UpsellBannerHeader variant render matches screenshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f2f4f6",
      "borderRadius": 8,
      "gap": 8,
      "paddingHorizontal": 16,
      "paddingVertical": 24,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#141618",
        "fontFamily": "EuclidCircularB-Bold",
        "fontSize": 18,
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 24,
      }
    }
  >
    you could earn
  </Text>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#1c8234",
        "fontFamily": "EuclidCircularB-Bold",
        "fontSize": 32,
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 40,
      }
    }
  >
    $454
  </Text>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#6a737d",
        "fontFamily": "EuclidCircularB-Regular",
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 22,
      }
    }
  >
    per year on your tokens
  </Text>
  <TouchableOpacity
    accessibilityRole="button"
    accessible={true}
    activeOpacity={1}
    onPress={[MockFunction]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "#0376c9",
        "borderRadius": 20,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#0376c9",
          "fontFamily": "EuclidCircularB-Medium",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      Earn 4.5%
    </Text>
  </TouchableOpacity>
</View>
`;
